tests/DeviceServiceTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dot\tests;
+
+use DeviceDetector\DeviceDetector;
+use Dot\UserAgentSniffer\Data\DeviceData;
+use Dot\UserAgentSniffer\Service\DeviceService;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class DeviceServiceTest extends TestCase
+{
+    private const TYPE_SMARTPHONE = 'smartphone';
+    private const TYPE_BROWSER    = 'browser';
+
+    private DeviceService $subject;
+
+    private DeviceDetector|MockObject $deviceDetectorMock;
+
+    private DeviceData|MockObject $deviceDataMock;
+
+    public function setUp(): void
+    {
+        parent::setUp();
```

Comment:
You can remove the call to `parent::setUp();`, because it does nothing.
The `setUp` method in `TestCase` is there only to make sure that the method is callable before each test, even if a test does not implement it.
In this case, you can remove the whole `setUp` method from your test.